spark/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala

```diff
@@ -125,6 +124,39 @@ object VacuumCommand extends VacuumCommandImpl with Serializable {
     }.toDF("path")
   }
 
+  def getFilesFromReservoir(basePath: String,
+                            partitionColumns: Seq[String],
+                            reservoir: DataFrame): Dataset[SerializableFileStatus] = {
+    val expectedSchema = StructType(
+      Seq(
+        StructField("path", StringType),
+        StructField("length", LongType),
+        StructField("isDir", BooleanType),
+        StructField("modificationTime", LongType)
+      ))
+
+    if(reservoir.schema != expectedSchema) {
+      throw new SparkException(s"reservoir schema should match ${expectedSchema.treeString}")
+    }
+
+    implicit val fileNameAndSizeEncoder
+    = org.apache.spark.sql.Encoders.product[SerializableFileStatus]
+
+    reservoir
+      .filter(s"path like '$basePath/%' ")
+      .selectExpr(s"replace(path, '$basePath/', '') as path", "length", "isDir", "modificationTime")
```

Comment:
The `replace` searches the entire string, which could lead to false matches if the path is local files (not a URI) and the base path is repeated later in the path. How about using `s"substr(path, ${basePath.length+2})` or something like that?